src/wasm.h

```diff
@@ -1720,33 +1722,41 @@ class Export {
   ExternalKind kind;
 };
 
+struct ElemSegment : Named {
+  enum Status { ACTIVE = 0, PASSIVE = 1, DECLARATIVE = 2 };
```

Comment:
Our convention has been to call this kind of enum `Kind` and to use use `PascalCase` rather than all-caps for the values.

Declarative segments are just for predeclaring what functions can be the operands of `ref.func`, right? Since they don't add any semantic value of their own, it might be simpler to not store them at all in the IR and to recreate them as part of Module writing. If we can avoid storing declarative segments, then we could just use the presence of `Name table` to differentiate between active and passive segments, so we wouldn't need this enum at all.

Reply:
Hey I like that idea a lot! But wouldn't that mean we have to loop over all `ref.func` expressions to reconstruct the declarative element segment? It's either that, or keeping track of all `ref.func` operands while constructing/modifying the IR, which I suspect is even more complicated.

Reply:
Yes, we would have to run an analysis pass over the module to find all the operands of `ref.func` instructions. We already do something similar to collect all the types that need to be emitted into the type section, so you could use something similar. Check out the implementation and uses of `ModuleUtils::collectHeapTypes`.

Reply:
Yeah great, I've already refactored the changes to remove `Table.segments` and also tried to eliminate all use cases of this `Status` enum. Reconstructing the declarative segment should be easy now.